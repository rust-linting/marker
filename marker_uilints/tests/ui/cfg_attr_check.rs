#![cfg_attr(marker, feature(register_tool))]
#![cfg_attr(marker, register_tool(marker))]

mod allow_with_simple_attr {
    #[cfg_attr(marker, allow(marker::item_with_test_name))]
    fn find_me_fn() {}
}

mod allow_with_crate_check_attr {
    #[cfg_attr(marker = "marker_uilints", allow(marker::item_with_test_name))]
    fn find_me_fn() {}
}

mod lint_with_unloaded_crate_attr {
    #[cfg_attr(marker = "some_unknown_crate_that_isnt_loaded", allow(marker::item_with_test_name))]
    fn find_me_fn() {}
}

mod unknown_lint_allow {
    #[cfg_attr(marker, allow(marker::some_unknown_lint_that_doesnt_exist))]
    fn foo() {}
}
