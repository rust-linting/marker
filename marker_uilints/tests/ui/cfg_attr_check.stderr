warning: unknown lint: `marker::some_unknown_lint_that_doesnt_exist`
  --> $DIR/cfg_attr_check.rs:20:30
   |
20 |     #[cfg_attr(marker, allow(marker::some_unknown_lint_that_doesnt_exist))]
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unknown_lints)]` on by default

warning: found a `fn` item with a test name
  --> $DIR/cfg_attr_check.rs:16:5
   |
16 |     fn find_me_fn() {}
   |     ^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(marker::item_with_test_name)]` on by default

warning: 2 warnings emitted

